plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/config/LibraryBuilder.java

```diff
@@ -0,0 +1,107 @@
+package com.google.cloud.tools.eclipse.appengine.libraries.config;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.InvalidRegistryObjectException;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.Filter;
+import com.google.cloud.tools.eclipse.appengine.libraries.Library;
+import com.google.cloud.tools.eclipse.appengine.libraries.LibraryFile;
+import com.google.cloud.tools.eclipse.appengine.libraries.MavenCoordinates;
+
+public class LibraryBuilder {
+
+  private static final String LIBRARY = "library";
+  private static final String ID = "id";
```

Comment:
An Eclipse convention is to prefix attributes with `ATTR_` and elements with `ELMT_` so that you can tell that they're `IConfigElement`-related.